src/Host/Client/Impl/Host/BrokerClientProxy.cs

```diff
@@ -26,10 +26,10 @@ internal sealed class BrokerClientProxy : IBrokerClient {
             broker.Dispose();
         }
 
-        public string Name => _broker.Name;
-        public bool IsRemote => _broker.IsRemote;
-        public Uri Uri => _broker.Uri;
-        public AboutHost AboutHost => _broker.AboutHost;
+        public string Name => _broker?.Name;
```

Comment:
Yes, that's true. In this case, we need to change `Dispose()` method:
```{csharp}
        public void Dispose() {
            var broker = Interlocked.Exchange(ref _broker, new NullBrokerClient());
            broker.Dispose();
        }
```

---
In reply to: [81584799](https://github.com/Microsoft/RTVS/pull/2416#discussion_r81584799) [](ancestors = 81584799)